src/deploy/osp_deployer/checkpoints.py

```diff
@@ -269,59 +273,123 @@ def verify_nodes_registered_in_ironic(self):
                           + len(setts.compute_nodes)
                           + len(setts.ceph_nodes)
                           + len(setts.computehci_nodes))
-
-        for node_type, nodes in setts.node_types_map.items():
-            logger.debug("Number of %s nodes: %s", node_type, str(len(nodes)))
-            expected_nodes += len(nodes)
-        if len(ls_nodes) != expected_nodes:
+        # hack, checking we have at least the correct number of core nodes
+        # TODO, we should get more specific for core node validation
+        if len(ls_nodes) < expected_nodes:
```

Comment:
Why can t we compare the exact number of expected core nodes ?

Reply:
It should be done, I was pushing it off as of now we only do openstack baremetal node list and count how many there are.  The validation will need to be updated to examine each baremetal node's properties, specifically:
 node['capabilities']['service_tag'] and validate the overcloud nodes defined in .properties have exact service tag matches in the ironic database.
I'll add that functionality and update the PR

Reply:
Done, will be reflected in next patch set.